test/BoardTests.cpp

```diff
@@ -64,3 +64,32 @@ TEST_F(BoardTests, checkJSONFileReaderExceptions)
                              jfr4.readRows(),
                              jfr4.readCols()), InvalidDimensions);
 }
+TEST_F(BoardTests, checkBoardArgumentBoard_WithGetBoardLinesFunction)
+{
+    std::vector<int> vec = {1,2};
+    std::vector<std::vector<int>> vec2 = {vec,vec};
+
+    Board board(3, 3, vec2, vec2);
+    ASSERT_EQ(board.getBoardLines().at(1)[1],FieldsEnum::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1)[0],FieldsEnum::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[1],FieldsEnum::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[0],FieldsEnum::UNKNOWN);
+}
+
+TEST_F(BoardTests, findLongestVectorOfCluesInRows)
+{
+    Board board(11,
+                13,
+                {{1},{3},{5},{7},{9},{11},{11},{11},{3,1,3},{3},{3},{5},{5}},
+                {{3},{5},{6},{2,7},{4,7},{13},{4,7},{2,7},{6},{5},{3}});
+    ASSERT_EQ(board.findLongestVectorInRows(),3);
```

Comment:
This test example is to complicated to see correct result on instant. consider making it shorter and clearer.

Reply:
Yes, you are right. I used in this test example of a ready solution for Logic images.

Reply:
done